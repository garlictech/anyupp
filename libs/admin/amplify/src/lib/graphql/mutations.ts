/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const createAdminUser = /* GraphQL */ `
  mutation CreateAdminUser(
    $input: CreateAdminUserInput!
    $condition: ModelAdminUserConditionInput
  ) {
    createAdminUser(input: $input, condition: $condition) {
      id
      name
      profileImage
      roles {
        role
        entities {
          chainId
          groupId
          unitId
        }
      }
      settings {
        selectedChainId
        selectedGroupId
        selectedUnitId
        selectedProductCategoryId
        selectedLanguage
        selectedHistoryDate
      }
      email
      phone
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateAdminUser = /* GraphQL */ `
  mutation UpdateAdminUser(
    $input: UpdateAdminUserInput!
    $condition: ModelAdminUserConditionInput
  ) {
    updateAdminUser(input: $input, condition: $condition) {
      id
      name
      profileImage
      roles {
        role
        entities {
          chainId
          groupId
          unitId
        }
      }
      settings {
        selectedChainId
        selectedGroupId
        selectedUnitId
        selectedProductCategoryId
        selectedLanguage
        selectedHistoryDate
      }
      email
      phone
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteAdminUser = /* GraphQL */ `
  mutation DeleteAdminUser(
    $input: DeleteAdminUserInput!
    $condition: ModelAdminUserConditionInput
  ) {
    deleteAdminUser(input: $input, condition: $condition) {
      id
      name
      profileImage
      roles {
        role
        entities {
          chainId
          groupId
          unitId
        }
      }
      settings {
        selectedChainId
        selectedGroupId
        selectedUnitId
        selectedProductCategoryId
        selectedLanguage
        selectedHistoryDate
      }
      email
      phone
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createStripeCard = /* GraphQL */ `
  mutation CreateStripeCard(
    $input: CreateStripeCardInput!
    $condition: ModelStripeCardConditionInput
  ) {
    createStripeCard(input: $input, condition: $condition) {
      brand
      checks {
        address_line1_check
        address_postal_code_check
        cvc_check
      }
      country
      last4
      exp_month
      exp_year
      fingerprint
      funding
      three_d_secure
      id
      object
      metadata {
        key
        value
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateStripeCard = /* GraphQL */ `
  mutation UpdateStripeCard(
    $input: UpdateStripeCardInput!
    $condition: ModelStripeCardConditionInput
  ) {
    updateStripeCard(input: $input, condition: $condition) {
      brand
      checks {
        address_line1_check
        address_postal_code_check
        cvc_check
      }
      country
      last4
      exp_month
      exp_year
      fingerprint
      funding
      three_d_secure
      id
      object
      metadata {
        key
        value
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteStripeCard = /* GraphQL */ `
  mutation DeleteStripeCard(
    $input: DeleteStripeCardInput!
    $condition: ModelStripeCardConditionInput
  ) {
    deleteStripeCard(input: $input, condition: $condition) {
      brand
      checks {
        address_line1_check
        address_postal_code_check
        cvc_check
      }
      country
      last4
      exp_month
      exp_year
      fingerprint
      funding
      three_d_secure
      id
      object
      metadata {
        key
        value
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createChain = /* GraphQL */ `
  mutation CreateChain(
    $input: CreateChainInput!
    $condition: ModelChainConditionInput
  ) {
    createChain(input: $input, condition: $condition) {
      id
      name
      description {
        en
        de
        hu
      }
      style {
        colors {
          backgroundLight
          backgroundDark
          borderLight
          borderDark
          disabled
          highlight
          indicator
          textLight
          textDark
        }
        images {
          header
          logo
        }
      }
      isActive
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateChain = /* GraphQL */ `
  mutation UpdateChain(
    $input: UpdateChainInput!
    $condition: ModelChainConditionInput
  ) {
    updateChain(input: $input, condition: $condition) {
      id
      name
      description {
        en
        de
        hu
      }
      style {
        colors {
          backgroundLight
          backgroundDark
          borderLight
          borderDark
          disabled
          highlight
          indicator
          textLight
          textDark
        }
        images {
          header
          logo
        }
      }
      isActive
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteChain = /* GraphQL */ `
  mutation DeleteChain(
    $input: DeleteChainInput!
    $condition: ModelChainConditionInput
  ) {
    deleteChain(input: $input, condition: $condition) {
      id
      name
      description {
        en
        de
        hu
      }
      style {
        colors {
          backgroundLight
          backgroundDark
          borderLight
          borderDark
          disabled
          highlight
          indicator
          textLight
          textDark
        }
        images {
          header
          logo
        }
      }
      isActive
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createGroup = /* GraphQL */ `
  mutation CreateGroup(
    $input: CreateGroupInput!
    $condition: ModelGroupConditionInput
  ) {
    createGroup(input: $input, condition: $condition) {
      id
      chainId
      name
      description {
        en
        de
        hu
      }
      currency
      address {
        address
        city
        country
        title
        postalCode
        location {
          lat
          lng
        }
      }
      email
      phone
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateGroup = /* GraphQL */ `
  mutation UpdateGroup(
    $input: UpdateGroupInput!
    $condition: ModelGroupConditionInput
  ) {
    updateGroup(input: $input, condition: $condition) {
      id
      chainId
      name
      description {
        en
        de
        hu
      }
      currency
      address {
        address
        city
        country
        title
        postalCode
        location {
          lat
          lng
        }
      }
      email
      phone
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteGroup = /* GraphQL */ `
  mutation DeleteGroup(
    $input: DeleteGroupInput!
    $condition: ModelGroupConditionInput
  ) {
    deleteGroup(input: $input, condition: $condition) {
      id
      chainId
      name
      description {
        en
        de
        hu
      }
      currency
      address {
        address
        city
        country
        title
        postalCode
        location {
          lat
          lng
        }
      }
      email
      phone
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createOrder = /* GraphQL */ `
  mutation CreateOrder(
    $input: CreateOrderInput!
    $condition: ModelOrderConditionInput
  ) {
    createOrder(input: $input, condition: $condition) {
      id
      created
      items {
        id
        created
        productName {
          en
          de
          hu
        }
        priceShown {
          currency
          pricePerUnit
          priceSum
          tax
          taxSum
        }
        productId
        quantity
        statusLog {
          userId
          status
          ts
        }
        variantId
        variantName {
          en
          de
          hu
        }
        laneId
      }
      paymentMethod
      staffId
      statusLog {
        userId
        status
        ts
      }
      sumPriceShown {
        currency
        pricePerUnit
        priceSum
        tax
        taxSum
      }
      takeAway
      userId
      place {
        seat
        table
      }
      paymentIntention
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateOrder = /* GraphQL */ `
  mutation UpdateOrder(
    $input: UpdateOrderInput!
    $condition: ModelOrderConditionInput
  ) {
    updateOrder(input: $input, condition: $condition) {
      id
      created
      items {
        id
        created
        productName {
          en
          de
          hu
        }
        priceShown {
          currency
          pricePerUnit
          priceSum
          tax
          taxSum
        }
        productId
        quantity
        statusLog {
          userId
          status
          ts
        }
        variantId
        variantName {
          en
          de
          hu
        }
        laneId
      }
      paymentMethod
      staffId
      statusLog {
        userId
        status
        ts
      }
      sumPriceShown {
        currency
        pricePerUnit
        priceSum
        tax
        taxSum
      }
      takeAway
      userId
      place {
        seat
        table
      }
      paymentIntention
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteOrder = /* GraphQL */ `
  mutation DeleteOrder(
    $input: DeleteOrderInput!
    $condition: ModelOrderConditionInput
  ) {
    deleteOrder(input: $input, condition: $condition) {
      id
      created
      items {
        id
        created
        productName {
          en
          de
          hu
        }
        priceShown {
          currency
          pricePerUnit
          priceSum
          tax
          taxSum
        }
        productId
        quantity
        statusLog {
          userId
          status
          ts
        }
        variantId
        variantName {
          en
          de
          hu
        }
        laneId
      }
      paymentMethod
      staffId
      statusLog {
        userId
        status
        ts
      }
      sumPriceShown {
        currency
        pricePerUnit
        priceSum
        tax
        taxSum
      }
      takeAway
      userId
      place {
        seat
        table
      }
      paymentIntention
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createProductCategory = /* GraphQL */ `
  mutation CreateProductCategory(
    $input: CreateProductCategoryInput!
    $condition: ModelProductCategoryConditionInput
  ) {
    createProductCategory(input: $input, condition: $condition) {
      id
      description {
        en
        de
        hu
      }
      image
      name {
        en
        de
        hu
      }
      position
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateProductCategory = /* GraphQL */ `
  mutation UpdateProductCategory(
    $input: UpdateProductCategoryInput!
    $condition: ModelProductCategoryConditionInput
  ) {
    updateProductCategory(input: $input, condition: $condition) {
      id
      description {
        en
        de
        hu
      }
      image
      name {
        en
        de
        hu
      }
      position
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteProductCategory = /* GraphQL */ `
  mutation DeleteProductCategory(
    $input: DeleteProductCategoryInput!
    $condition: ModelProductCategoryConditionInput
  ) {
    deleteProductCategory(input: $input, condition: $condition) {
      id
      description {
        en
        de
        hu
      }
      image
      name {
        en
        de
        hu
      }
      position
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createChainProduct = /* GraphQL */ `
  mutation CreateChainProduct(
    $input: CreateChainProductInput!
    $condition: ModelChainProductConditionInput
  ) {
    createChainProduct(input: $input, condition: $condition) {
      id
      description {
        en
        de
        hu
      }
      extends
      image
      isVisible
      tax
      name {
        en
        de
        hu
      }
      position
      productCategoryId
      laneId
      productType
      variants {
        variantName {
          en
          de
          hu
        }
        pack {
          size
          unit
        }
        refGroupPrice
        isAvailable
        price
        availabilities {
          type
          dayFrom
          dayTo
          timeFrom
          timeTo
          price
        }
        availableFrom
        position
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateChainProduct = /* GraphQL */ `
  mutation UpdateChainProduct(
    $input: UpdateChainProductInput!
    $condition: ModelChainProductConditionInput
  ) {
    updateChainProduct(input: $input, condition: $condition) {
      id
      description {
        en
        de
        hu
      }
      extends
      image
      isVisible
      tax
      name {
        en
        de
        hu
      }
      position
      productCategoryId
      laneId
      productType
      variants {
        variantName {
          en
          de
          hu
        }
        pack {
          size
          unit
        }
        refGroupPrice
        isAvailable
        price
        availabilities {
          type
          dayFrom
          dayTo
          timeFrom
          timeTo
          price
        }
        availableFrom
        position
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteChainProduct = /* GraphQL */ `
  mutation DeleteChainProduct(
    $input: DeleteChainProductInput!
    $condition: ModelChainProductConditionInput
  ) {
    deleteChainProduct(input: $input, condition: $condition) {
      id
      description {
        en
        de
        hu
      }
      extends
      image
      isVisible
      tax
      name {
        en
        de
        hu
      }
      position
      productCategoryId
      laneId
      productType
      variants {
        variantName {
          en
          de
          hu
        }
        pack {
          size
          unit
        }
        refGroupPrice
        isAvailable
        price
        availabilities {
          type
          dayFrom
          dayTo
          timeFrom
          timeTo
          price
        }
        availableFrom
        position
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createUnit = /* GraphQL */ `
  mutation CreateUnit(
    $input: CreateUnitInput!
    $condition: ModelUnitConditionInput
  ) {
    createUnit(input: $input, condition: $condition) {
      id
      groupId
      isActive
      isAcceptingOrders
      name
      description {
        en
        de
        hu
      }
      paymentModes {
        name
        caption
        method
      }
      floorMap {
        w
        h
        objects {
          id
          t
          c
          w
          h
          r
          a
          x
          y
          tID
          sID
        }
      }
      lanes {
        id
        name
        color
      }
      open {
        from
        to
      }
      openingHours {
        mon {
          from
          to
        }
        tue {
          from
          to
        }
        wed {
          from
          to
        }
        thu {
          from
          to
        }
        fri {
          from
          to
        }
        sat {
          from
          to
        }
        sun {
          from
          to
        }
        override {
          date
          from
          to
        }
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateUnit = /* GraphQL */ `
  mutation UpdateUnit(
    $input: UpdateUnitInput!
    $condition: ModelUnitConditionInput
  ) {
    updateUnit(input: $input, condition: $condition) {
      id
      groupId
      isActive
      isAcceptingOrders
      name
      description {
        en
        de
        hu
      }
      paymentModes {
        name
        caption
        method
      }
      floorMap {
        w
        h
        objects {
          id
          t
          c
          w
          h
          r
          a
          x
          y
          tID
          sID
        }
      }
      lanes {
        id
        name
        color
      }
      open {
        from
        to
      }
      openingHours {
        mon {
          from
          to
        }
        tue {
          from
          to
        }
        wed {
          from
          to
        }
        thu {
          from
          to
        }
        fri {
          from
          to
        }
        sat {
          from
          to
        }
        sun {
          from
          to
        }
        override {
          date
          from
          to
        }
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteUnit = /* GraphQL */ `
  mutation DeleteUnit(
    $input: DeleteUnitInput!
    $condition: ModelUnitConditionInput
  ) {
    deleteUnit(input: $input, condition: $condition) {
      id
      groupId
      isActive
      isAcceptingOrders
      name
      description {
        en
        de
        hu
      }
      paymentModes {
        name
        caption
        method
      }
      floorMap {
        w
        h
        objects {
          id
          t
          c
          w
          h
          r
          a
          x
          y
          tID
          sID
        }
      }
      lanes {
        id
        name
        color
      }
      open {
        from
        to
      }
      openingHours {
        mon {
          from
          to
        }
        tue {
          from
          to
        }
        wed {
          from
          to
        }
        thu {
          from
          to
        }
        fri {
          from
          to
        }
        sat {
          from
          to
        }
        sun {
          from
          to
        }
        override {
          date
          from
          to
        }
      }
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const createUser = /* GraphQL */ `
  mutation CreateUser(
    $input: CreateUserInput!
    $condition: ModelUserConditionInput
  ) {
    createUser(input: $input, condition: $condition) {
      id
      name
      email
      phone
      profileImage
      login
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const updateUser = /* GraphQL */ `
  mutation UpdateUser(
    $input: UpdateUserInput!
    $condition: ModelUserConditionInput
  ) {
    updateUser(input: $input, condition: $condition) {
      id
      name
      email
      phone
      profileImage
      login
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
export const deleteUser = /* GraphQL */ `
  mutation DeleteUser(
    $input: DeleteUserInput!
    $condition: ModelUserConditionInput
  ) {
    deleteUser(input: $input, condition: $condition) {
      id
      name
      email
      phone
      profileImage
      login
      _version
      _deleted
      _lastChangedAt
      createdAt
      updatedAt
    }
  }
`;
