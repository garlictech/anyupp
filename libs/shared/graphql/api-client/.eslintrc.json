{
  "extends": "../../../../.eslintrc.json",
  "ignorePatterns": [
    "!**/*"
  ],
  "rules": {},
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "plugins": [
        "@typescript-eslint"
      ],
      "parserOptions": {
        "project": [
          "libs/shared/graphql/api-client/tsconfig.*?.json"
        ]
      }
    }
  ]
}
