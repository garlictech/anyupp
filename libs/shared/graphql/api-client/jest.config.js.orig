module.exports = {
  displayName: 'shared-graphql-api-client',
  preset: '../../../../jest.preset.js',
  globals: {
<<<<<<< HEAD
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
||||||| dbeda1c5
    'ts-jest': {
      tsConfig: '<rootDir>/tsconfig.spec.json',
    },
=======
    'ts-jest': { tsconfig: '<rootDir>/tsconfig.spec.json' },
>>>>>>> dev
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/shared/graphql/api-client',
};
