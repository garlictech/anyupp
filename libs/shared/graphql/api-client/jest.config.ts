export default {
  displayName: 'shared-graphql-api-client',

  globals: {
    'ts-jest': { tsconfig: '<rootDir>/tsconfig.spec.json' },
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/shared/graphql/api-client',
  preset: '../../../../jest.preset.js',
};
