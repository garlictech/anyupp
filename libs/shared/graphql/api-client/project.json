{
  "root": "libs/shared/graphql/api-client",
  "sourceRoot": "libs/shared/graphql/api-client/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/shared/graphql/api-client/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/shared/graphql/api-client"],
      "options": {
        "jestConfig": "libs/shared/graphql/api-client/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["layer:utils", "platform:nodejs"]
}
