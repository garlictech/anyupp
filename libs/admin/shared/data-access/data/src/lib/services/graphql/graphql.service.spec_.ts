import { TestBed } from '@angular/core/testing';

import { GraphQLService } from './graphql.service';

xdescribe('OrderService', (): void => {
  let service: GraphQLService;

  beforeEach((): void => {
    TestBed.configureTestingModule({});
    service = TestBed.inject(GraphQLService);
  });

  it('should be created', (): void => {
    expect(service).toBeTruthy();
  });
});
