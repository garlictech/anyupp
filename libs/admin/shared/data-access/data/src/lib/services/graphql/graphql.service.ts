import { Apollo } from 'apollo-angular';
import { IKeyValueObject } from '@bgap/shared/types';

import { Injectable } from '@angular/core';
import { CreateAdminUser, UpdateAdminUser, GetAdminUsers } from '@bgap/api/graphql/schema';

@Injectable({
  providedIn: 'root',
})
export class GraphQLService {
  constructor(private _apollo: Apollo) {}

  //
  // ADMIN USER
  //

  public getAdminUsers() {
    return this._apollo.query({
      query: GetAdminUsers
    });
  }

  public createAdminUser(data: IKeyValueObject) {
    return this._apollo.mutate({
      mutation: CreateAdminUser,
      variables: {
        data,
      },
    });
  }

  public updateAdminUser(id: string, data: IKeyValueObject) {
    return this._apollo.mutate({
      mutation: UpdateAdminUser,
      variables: {
        data,
        id,
      },
    });
  }
}
