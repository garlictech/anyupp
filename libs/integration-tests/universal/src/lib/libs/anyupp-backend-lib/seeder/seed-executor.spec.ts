import {
  createTestAdminRoleContext,
  createTestRoleContext,
  createTestUnitProduct,
  seedAdminUser,
  seedBusinessData,
} from '@bgap/anyupp-backend-lib';
import {
  createTestChain,
  createTestChainProduct,
} from '@bgap/anyupp-backend-lib';
import { AWS_CRUD_CONFIG } from '@bgap/shared/graphql/api-client';
import { AWSError } from 'aws-sdk';
import { of, throwError } from 'rxjs';
import { catchError, switchMap, delay } from 'rxjs/operators';

describe('Seeder test', () => {
  describe('DB seeder', () => {
    it.skip('should createTestChain work', done => {
      createTestChain(1).subscribe({
        next(result: any) {
          expect(result).toHaveProperty('createChain');
          const {
            createdAt,
            updatedAt,
            ...justCreatedObject
          } = result.createChain;
          expect(createdAt).not.toBeUndefined();
          expect(updatedAt).not.toBeUndefined();
          expect(justCreatedObject).toMatchSnapshot();
          done();
        },
      });
    }, 25000);

    it.skip('should createTestChainProduct work', done => {
      createTestChainProduct(1, 1, 1).subscribe({
        next(result: any) {
          expect(result).toHaveProperty('createChainProduct');
          const {
            createdAt,
            updatedAt,
            ...justCreatedObject
          } = result.createChainProduct;
          expect(createdAt).not.toBeUndefined();
          expect(updatedAt).not.toBeUndefined();
          expect(justCreatedObject).toMatchSnapshot();
          done();
        },
      });
    }, 25000);

    it.skip('should createTestUnitProduct work', done => {
      createTestUnitProduct(1, 1, 1, 1, 1).subscribe({
        next(result: any) {
          expect(result).toHaveProperty('unitProduct');
          expect(result).toHaveProperty('generatedProduct');
          const {
            createdAt: ca01,
            updatedAt: ua01,
            ...justCreatedUnitProduct
          } = result.unitProduct;
          const {
            createdAt: ca02,
            updatedAt: ua02,
            ...justCreatedGeneratedObject
          } = result.generatedProduct;
          expect(ca01).not.toBeUndefined();
          expect(ca02).not.toBeUndefined();
          expect(ua01).not.toBeUndefined();
          expect(ua02).not.toBeUndefined();
          expect(justCreatedUnitProduct).toMatchSnapshot('Unit Product');
          expect(justCreatedGeneratedObject).toMatchSnapshot(
            'Generated Product',
          );
          done();
        },
      });
    }, 25000);

    it.skip('should the seedBusinessData function run successfully', done => {
      seedBusinessData('USER_ID').subscribe({
        complete() {
          done();
        },
      });
    }, 25000);
    it.skip('should the createTestRoleContext function run successfully', done => {
      createTestRoleContext(1, 1, 1, 1).subscribe({
        complete() {
          done();
        },
      });
    }, 25000);
    it.skip('should the createTestAdminRoleContext function run successfully', done => {
      createTestAdminRoleContext(
        1,
        1,
        '!!! REQUIRES EXISTING USER ID !!!',
      ).subscribe({
        complete() {
          done();
        },
      });
    }, 25000);
  });

  describe('AdminUser seeder', () => {
    it.skip('Admin user create', done => {
      seedAdminUser(AWS_CRUD_CONFIG.aws_user_pools_id).subscribe({
        complete() {
          done();
        },
      });
    }, 25000);
  });

  it.skip('should run complete seeder', done => {
    seedAdminUser(AWS_CRUD_CONFIG.aws_user_pools_id)
      .pipe(
        delay(2000),
        switchMap(userId => seedBusinessData(userId)),
        catchError((error: AWSError) => {
          console.error('SEEDING ERROR', JSON.stringify(error, undefined, 2));
          return throwError('SUCCESS');
        }),
      )
      .subscribe({
        complete() {
          done();
        },
      });
  }, 25000);
});
