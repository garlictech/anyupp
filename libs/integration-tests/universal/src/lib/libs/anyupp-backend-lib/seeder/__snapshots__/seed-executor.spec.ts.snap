// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seeder test DB seeder should createTestUnitProduct work: Generated Product 1`] = `
Object {
  "__typename": "GeneratedProduct",
  "description": Object {
    "__typename": "LocalizedItem",
    "de": null,
    "en": "GENERATED_PRODUCT DESCRIPTION unit_product_c1_g1_1_id_seeded",
    "hu": "GENERALT_TERMEK  LEIRAS unit_product_c1_g1_1_id_seeded",
  },
  "id": "unit_product_c1_g1_1_id_seeded",
  "image": "public/seed/Weizenbier.jpeg",
  "name": Object {
    "__typename": "LocalizedItem",
    "de": null,
    "en": "GENERATED_PRODUCT_unit_product_c1_g1_1_id_seeded",
    "hu": "GENERALT_TERMEK_unit_product_c1_g1_1_id_seeded",
  },
  "position": 1,
  "productCategoryId": "product_category_c1_1_id_seeded",
  "productType": "PRODUCT_TYPE",
  "tax": 1,
  "unitId": "unit_c1_g1_1_id_seeded",
  "variants": Array [
    Object {
      "__typename": "GeneratedProductVariant",
      "id": "generatedProductVariant_id_unit_product_c1_g1_1_id_seeded",
      "pack": Object {
        "__typename": "ProductVariantPack",
        "size": 1,
        "unit": "UNIT",
      },
      "position": 1,
      "price": 1,
      "variantName": Object {
        "__typename": "LocalizedItem",
        "de": null,
        "en": "VARIANT_NAME_unit_product_c1_g1_1_id_seeded",
        "hu": "VARIANT_NAME_unit_product_c1_g1_1_id_seeded",
      },
    },
    Object {
      "__typename": "GeneratedProductVariant",
      "id": "generatedProductVariant_id_unit_product_c1_g1_1_id_seeded",
      "pack": Object {
        "__typename": "ProductVariantPack",
        "size": 2,
        "unit": "UNIT",
      },
      "position": 2,
      "price": 2,
      "variantName": Object {
        "__typename": "LocalizedItem",
        "de": null,
        "en": "VARIANT_NAME_unit_product_c1_g1_1_id_seeded",
        "hu": "VARIANT_NAME_unit_product_c1_g1_1_id_seeded",
      },
    },
  ],
}
`;

exports[`Seeder test DB seeder should createTestUnitProduct work: Unit Product 1`] = `
Object {
  "__typename": "UnitProduct",
  "chainId": "chain_1_id_seeded",
  "groupId": "group_c1_1_id_seeded",
  "groupProduct": null,
  "id": "unit_product_c1_g1_1_id_seeded",
  "isVisible": true,
  "laneId": "lane_c1_g1_u1_1_id_seeded",
  "parentId": "group_product_c1_g1_1_id_seeded",
  "position": 1,
  "takeaway": false,
  "unitId": "unit_c1_g1_1_id_seeded",
  "variants": Array [
    Object {
      "__typename": "ProductVariant",
      "availabilities": Array [
        Object {
          "__typename": "Availability",
          "dayFrom": "",
          "dayTo": "",
          "price": 60,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "chain_product_variant_c1_p1_1_id_seeded",
      "isAvailable": true,
      "pack": Object {
        "__typename": "ProductVariantPack",
        "size": 2,
        "unit": "dl",
      },
      "position": 1,
      "price": 11,
      "refGroupPrice": null,
      "variantName": Object {
        "__typename": "LocalizedItem",
        "de": null,
        "en": "glass",
        "hu": "poh√°r",
      },
    },
  ],
}
`;
